holidays/countries/thailand.py

```diff
@@ -474,14 +474,19 @@ def _populate_public_holidays(self):
 
         if self._year >= 2017:
             k_bhumibol_memorial = (
-                # Anniversary for the Death of King Bhumibol Adulyadej the Great.
-                tr(
-                    "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
-                    "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
+                # HM King Bhumibol Adulyadej Memorial Day.
+                tr("วันนวมินทรมหาราช")
+                if self._year >= 2023
+                else (
+                    # Anniversary for the Death of King Bhumibol Adulyadej the Great.
+                    tr(
+                        "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
+                        "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
+                    )
+                    if self._year >= 2019
+                    # Anniversary for the Death of King Bhumibol Adulyadej.
+                    else tr("วันคล้ายวันสวรรคตพระบาทสมเด็จพระปรมินทรมหาภูมิพลอดุลยเดช บรมนาถบพิตร")
                 )
```

Comment:
I feel this nesting is a bit too much. I suggest using this syntax for simple single level if/else cases:

```
if self._year >= 2023:
  # HM King Bhumibol Adulyadej Memorial Day.
  k_bhumibol_memorial = tr("วันนวมินทรมหาราช")
elif self._year >= 2019:
    # Anniversary for the Death of King Bhumibol Adulyadej the Great.
    k_bhumibol_memorial = tr(
        "วันคล้ายวันสวรรคตพระบาทสมเด็จพระบรมชนกาธิเบศร "
        "มหาภูมิพลอดุลยเดชมหาราช บรมนาถบพิตร"
    )
else:
  # Anniversary for the Death of King Bhumibol Adulyadej.
  k_bhumibol_memorial = tr(
      "วันคล้ายวันสวรรคตพระบาทสมเด็จพระปรมินทรมหาภูมิพลอดุลยเดช บรมนาถบพิตร"
  )
```

Reply:
Fixed.

On minor note I'm pretty sure Netherland and its oversee territories are also got this format, so we should take a look at them later on.